Add "clean" script.

arthurevans opened the pull request:
This leaves somewhat fewer files to update when renaming the element. 

It does leave behind the my-element.bundled.js file in /docs/, since that is checked in. 

The vast number of references left in the files occur in /docs/ and /docs-src/. It seems like /docs/ should be entirely generated files, but it's checked in. It makes the update quite messy, but I don't see any way to address that in this PR.

justinfagnani commented:
GitHub's choices around doc locations make this a bit difficult. If we don't want generated docs checked in I think we'll need a script to generate them to gh-pages and push that branch. There are nice reasons to do that - like decoupling master from live docs - I've just never set that up before.

justinfagnani commented on package.json:
```diff
@@ -8,6 +8,7 @@
   "scripts": {
     "build": "tsc",
     "build:watch": "tsc --watch",
+    "clean": "rm -rf lib test my-element.{d.ts,d.ts.map,js,js.map}",
```
I'm not sure we want to remove all of test. I often have non-.ts files (like .html and data files) checked in there.

Could we use `git clean` here instead?

arthurevans replied:
We could--although I think it'll remove `node_modules` as well, which is not ideal.

arthurevans replied:
I looked at the .gitignore file when deciding what to remove, but clearly if there are non-generated files in test, we shouldn't remove the directory.

arthurevans replied:
Also, if you add files to `test` and run `npm clean` before committing them, `git clean` would remove those, too, which might be a little surprising. Perhaps best to explicitly remove just the generated files?

arthurevans commented:
Note that we aren't currently publishing GitHub pages for this repo.

arthurevans commented:
My inclination is to remove the `/docs/` folder from the repo and leave publishing the docs as an exercise for the user. Checking in `/docs/` and `custom-elements.json` here makes it more complete as an element, but doesn't really add to its utility as a template.

Alternately, we could:

- Add a "clean:all" script that removes the docs folder, or 
- Add "rm -rf docs" to the beginning of the "/docs/" script. 

Either of these would let people rename the element without having an extra `my-element.bundled.js` file sitting around in /docs/.

arthurevans commented:
Now only removing the generated files in test.

I added a step to clean the `/docs/` directory, and realized there were actually some static files in there that _didn't_ come from `/docs-src/`. I moved those so you can safely remove the docs directory, which feels less surprising.

arthurevans commented:
PTAL
